main/nucleus/core/bootstrap/src/test/java/com/sun/enterprise/glassfish/bootstrap/MainHelperTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of either the GNU
+ * General Public License Version 2 only ("GPL") or the Common Development
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * https://glassfish.dev.java.net/public/CDDL+GPL_1_1.html
+ * or packager/legal/LICENSE.txt.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at packager/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * Oracle designates this particular file as subject to the "Classpath"
+ * exception as provided by Oracle in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+
+package com.sun.enterprise.glassfish.bootstrap;
+
+/**
+ * Created by kokil on 5/18/17.
+ */
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import com.sun.enterprise.glassfish.bootstrap.MainHelper;
+import java.io.*;
+import java.util.*;
+
+public class MainHelperTest {
+    /* This test is used to test the regex pattern of "parseAsEnv" method of "MainHelper.java"*/
+    @Test
+    public void parseAsEnvTest() {
+        try {
+            Properties asenvProps = new Properties();
+            String currentDir = System.getProperty("user.dir");
```

Comment:
This seems to depend on the test files being in a certain directory that corresponds to the user.dir System property, but it's not clear to me how that's set up or what guarantees that.  Wouldn't it be better to create a temporary directory, create a file there, write the test cases into the file, and then pass the temporary directory to the parseAsEnv method?  Or, if there's some other guarantee that user.dir wil be set properly, document how that works.